In [ ]:
import pandas as pd
df = pd.read_csv('assignment_1-python (1).csv')
df

date      region        city      market  latitude  longitude  \
0    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
1    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
2    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
3    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
4    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
..          ...         ...         ...         ...       ...        ...   
349  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
350  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
351  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
352  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
353  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   

                  category                        commodity   unit priceflag  \
0       cereals and tubers                            Bread     KG    actual   
1       cereals and tubers           Potatoes (medium size)     KG    actual   
2       cereals and tubers     Rice (small grain, imported)  25 KG    actual   
3       cereals and tubers  Wheat flour (locally processed)  50 KG    actual   
4      meat, fish and eggs                             Eggs   2 KG    actual   
..                     ...                              ...    ...       ...   
349  vegetables and fruits                      Cauliflower     KG    actual   
350  vegetables and fruits           Cucumbers (greenhouse)     KG    actual   
351  vegetables and fruits                Eggplants (large)     KG    actual   
352  vegetables and fruits              Onions (dry, local)     KG    actual   
353  vegetables and fruits            Tomatoes (greenhouse)     KG    actual   

    pricetype currency   price  usdprice  
0      Retail      ILS    2.50    0.7934  
1      Retail      ILS    1.94    0.6157  
2      Retail      ILS  152.31   48.3351  
3      Retail      ILS   90.00   28.5612  
4      Retail      ILS   11.72    3.7193  
..        ...      ...     ...       ...  
349    Retail      ILS    2.95    0.8626  
350    Retail      ILS    1.83    0.5351  
351    Retail      ILS    1.95    0.5702  
352    Retail      ILS    2.06    0.6023  
353    Retail      ILS    2.92    0.8538  

[354 rows x 14 columns]

As we see here I read the file

In [ ]:
price_commodity=df[['commodity','price']]
price_commodity

commodity   price
0                              Bread    2.50
1             Potatoes (medium size)    1.94
2       Rice (small grain, imported)  152.31
3    Wheat flour (locally processed)   90.00
4                               Eggs   11.72
..                               ...     ...
349                      Cauliflower    2.95
350           Cucumbers (greenhouse)    1.83
351                Eggplants (large)    1.95
352              Onions (dry, local)    2.06
353            Tomatoes (greenhouse)    2.92

[354 rows x 2 columns]

I want only Commodity and price coulmns in this section

In [ ]:
most_expensive=price_commodity.groupby('price').max()
most_expensive

commodity
price                               
1.30                            Salt
1.48                Water (drinking)
1.50                            Salt
1.53             Onions (dry, local)
1.56                Water (drinking)
...                              ...
148.85  Rice (small grain, imported)
150.00  Rice (small grain, imported)
152.31  Rice (small grain, imported)
153.46  Rice (small grain, imported)
155.00  Rice (small grain, imported)

[231 rows x 1 columns]

In [ ]:
most_expensive.tail(1)

commodity
price                              
155.0  Rice (small grain, imported)

This is the most expensive comdity with price of 155.0

In [ ]:
df['currency_rate'] = df['price'] / df['usdprice']
df

date      region        city      market  latitude  longitude  \
0    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
1    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
2    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
3    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
4    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
..          ...         ...         ...         ...       ...        ...   
349  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
350  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
351  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
352  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
353  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   

                  category                        commodity   unit priceflag  \
0       cereals and tubers                            Bread     KG    actual   
1       cereals and tubers           Potatoes (medium size)     KG    actual   
2       cereals and tubers     Rice (small grain, imported)  25 KG    actual   
3       cereals and tubers  Wheat flour (locally processed)  50 KG    actual   
4      meat, fish and eggs                             Eggs   2 KG    actual   
..                     ...                              ...    ...       ...   
349  vegetables and fruits                      Cauliflower     KG    actual   
350  vegetables and fruits           Cucumbers (greenhouse)     KG    actual   
351  vegetables and fruits                Eggplants (large)     KG    actual   
352  vegetables and fruits              Onions (dry, local)     KG    actual   
353  vegetables and fruits            Tomatoes (greenhouse)     KG    actual   

    pricetype currency   price  usdprice  currency_rate  
0      Retail      ILS    2.50    0.7934       3.150996  
1      Retail      ILS    1.94    0.6157       3.150885  
2      Retail      ILS  152.31   48.3351       3.151126  
3      Retail      ILS   90.00   28.5612       3.151128  
4      Retail      ILS   11.72    3.7193       3.151131  
..        ...      ...     ...       ...            ...  
349    Retail      ILS    2.95    0.8626       3.419893  
350    Retail      ILS    1.83    0.5351       3.419922  
351    Retail      ILS    1.95    0.5702       3.419853  
352    Retail      ILS    2.06    0.6023       3.420222  
353    Retail      ILS    2.92    0.8538       3.420005  

[354 rows x 15 columns]

This is the exchange rate between the local currency and the US dollar (USD)

In [ ]:
df['unified_price'] = None
df['unified_price'] = df.apply(
    lambda row: row['usdprice'] if 'KG' in row['unit'] or 'kg' in row['unit'] or 'L' in row['unit'] or 'l' in row['unit']
    else row['usdprice'] * 1000 if 'g' in row['unit']
    else row['usdprice'] * 1000 if 'ml' in row['unit']
    else None, axis=1)
df

date      region        city      market  latitude  longitude  \
0    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
1    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
2    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
3    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
4    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
..          ...         ...         ...         ...       ...        ...   
349  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
350  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
351  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
352  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
353  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   

                  category                        commodity   unit priceflag  \
0       cereals and tubers                            Bread     KG    actual   
1       cereals and tubers           Potatoes (medium size)     KG    actual   
2       cereals and tubers     Rice (small grain, imported)  25 KG    actual   
3       cereals and tubers  Wheat flour (locally processed)  50 KG    actual   
4      meat, fish and eggs                             Eggs   2 KG    actual   
..                     ...                              ...    ...       ...   
349  vegetables and fruits                      Cauliflower     KG    actual   
350  vegetables and fruits           Cucumbers (greenhouse)     KG    actual   
351  vegetables and fruits                Eggplants (large)     KG    actual   
352  vegetables and fruits              Onions (dry, local)     KG    actual   
353  vegetables and fruits            Tomatoes (greenhouse)     KG    actual   

    pricetype currency   price  usdprice  currency_rate  unified_price  
0      Retail      ILS    2.50    0.7934       3.150996         0.7934  
1      Retail      ILS    1.94    0.6157       3.150885         0.6157  
2      Retail      ILS  152.31   48.3351       3.151126        48.3351  
3      Retail      ILS   90.00   28.5612       3.151128        28.5612  
4      Retail      ILS   11.72    3.7193       3.151131         3.7193  
..        ...      ...     ...       ...            ...            ...  
349    Retail      ILS    2.95    0.8626       3.419893         0.8626  
350    Retail      ILS    1.83    0.5351       3.419922         0.5351  
351    Retail      ILS    1.95    0.5702       3.419853         0.5702  
352    Retail      ILS    2.06    0.6023       3.420222         0.6023  
353    Retail      ILS    2.92    0.8538       3.420005         0.8538  

[354 rows x 16 columns]

As we see here  unified_price column  represents the price in USD per kilogram for commodities measured by weight and the price per liter for commodities measured by volume

In [ ]:
weight_units = ['KG', 'kg', 'g']
weight_commodities = df[df['unit'].isin(weight_units)]
max_weight_index = weight_commodities['unified_price'].idxmax()
max_weight_price = weight_commodities.loc[max_weight_index]
max_weight_price_info = {max_weight_price['commodity'],max_weight_price['unified_price']}

In [ ]:
max_weight_price_info

{17.3144, 'Meat (goat, with bones)'}

Max price for commodities measured by weight is Meat(goat,with bones) with price of 17.3144

In [ ]:
volume_units = ['L', 'l', 'ml']
volume_commodities = df[df['unit'].isin(volume_units)]
max_volume_index = volume_commodities['unified_price'].idxmax()
max_volume_price = volume_commodities.loc[max_volume_index]
max_volume_price_info = {max_volume_price['commodity'],max_volume_price['unified_price']
}

In [ ]:
max_volume_price_info

{2.0459, 'Fuel (petrol-gasoline)'}

Max price for commodities measured by volume is Fuel (petrol-gasoline) with price of 2.0459